instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/instrumentation.rb

```diff
@@ -41,31 +41,33 @@ def gem_version
 
         def require_dependencies
           require_relative 'handler'
-          require_relative 'services'
           require_relative 'message_attributes'
           require_relative 'messaging_helper'
         end
 
-        def add_plugin(*targets)
+        def add_plugins(*targets)
           targets.each { |klass| klass.add_plugin(AwsSdk::Plugin) }
         end
 
-        def loaded_constants
-          # Cross-check services against loaded AWS constants
-          # Module#const_get can return a constant from ancestors when there's a miss.
-          # If this conincidentally matches another constant, it will attempt to patch
-          # the wrong constant, resulting in patch failure.
-          available_services = ::Aws.constants & SERVICES.map(&:to_sym)
-          available_services.each_with_object([]) do |service, constants|
-            next if ::Aws.autoload?(service)
+        def loaded_service_clients
+          ::Aws.constants.each_with_object([]) do |c, constants|
+            m = ::Aws.const_get(c)
+            next unless loaded_service?(c, m)
 
             begin
-              constants << ::Aws.const_get(service, false).const_get(:Client, false)
+              constants << m.const_get(:Client)
             rescue StandardError => e
               OpenTelemetry.logger.warn("Constant could not be loaded: #{e}")
             end
           end
         end
+
+        def loaded_service?(constant, service_module)
+          !::Aws.autoload?(constant) &&
+            service_module.is_a?(Module) &&
+            service_module.const_defined?(:Client) &&
+            (service_module.const_get(:Client).superclass == Seahorse::Client::Base || Aws::Client)
```

Comment:
Hi! That's a great question. I think it's probably worthwhile for me to update this method with some comments of context.

Yeah, basically - we want to use this method to be compatible with MV 2 and 3 of the Ruby SDK. Both of these two MVs works a bit differently on how service clients are created underneath. V3 service client's superclass is `Seahorse::Client::Base` whereas V2 service client's superclass is `Aws::Client` - hence the conditional.

Reply:
Ahh, okay. Thank you! The comments you added are helpful!

I believe the current condition is evaluated like:
`((service_module.const_get(:Client).superclass == Seahorse::Client::Base) || Aws::Client)`

so whenever `Seahorse::Client::Base` is not the client superclass, the `Aws::Client` constant is returned instead of a boolean. `Aws::Client` would also be returned for constants without the right superclass. 

For example:
```ruby
def test(var)
  var == 'b' || 'a'
end

test('b')
=> true
test('a')
=> 'a'
test('c')
=> 'a'
```

I think what you may want is: 
`service_module.const_get(:Client).superclass == Seahorse::Client::Base || service_module.const_get(:Client).superclass == Aws::Client`

or maybe something like:
`[Seahorse::Client::Base, Aws::Client].include?(service_module.const_get(:Client).superclass)`

Reply:
Ahhh, thanks for this catch. You are correct. 😆
I'll push a change right now!

Reply:
Thank you! Change looks great, though I think something else might need to be updated for the tests? 
https://github.com/open-telemetry/opentelemetry-ruby-contrib/actions/runs/10838057325/job/30082448049?pr=1150#step:3:575

Reply:
Ahh, I see what happened here (it was due to how V3 vs V2 works again 😔). 
Hopefully, my update did the trick.